### PR TITLE
Rebuild Custom HTML Widget to eliminate reliance on `backbone.js` in favor of PHP

### PR DESCRIPTION
**Pull request by KTS915**
The Custom HTML widget was created in WP 4.9, when the old Text widget was split into two: a TinyMCE widget and this one. The old text widget had been based on PHP. For no good reason, the new Custom HTML widget was based on a `backbone.js` template, necessitating a JavaScript file of over 17kB to make it work.

This PR eliminates that insanity and bases the Custom HTML widget on PHP. This results in the relevant JavaScript file now being only 1.5kB in size, and even the widget's PHP file is smaller. This PR also makes the widget significantly more robust, since it won't stop functioning simply because there's a JavaScript error somewhere else on the page. It is also much easier to maintain, since not only is the code much less but it's also far easier to understand.

I have tested this PR on my localhost testing site. It seems to work both on the `widgets.php` page and in the Customizer.

### CODE REVIEW

File: src/wp-includes/widgets/class-wp-widget-custom-html.php [hunk 1]
```diff
@@ -202,44 +196,22 @@ public function update( $new_instance, $old_instance ) {
...
 	}
 
 	/**
 	 * Outputs the Custom HTML widget settings form.
 	 *
-	 * @since 4.8.1
-	 * @since 4.9.0 The form contains only hidden sync inputs. For the control UI, see `WP_Widget_Custom_HTML::render_control_template_scripts()`.
+	 * @since CP-2.3.0
 	 *
 	 * @see WP_Widget_Custom_HTML::render_control_template_scripts()
```
Comment (mattyrob) [inline src/wp-includes/widgets/class-wp-widget-custom-html.php:216]: This line reference a function that has been removed.

File: src/wp-includes/widgets/class-wp-widget-custom-html.php [hunk 2]
```diff
@@ -202,44 +196,22 @@ public function update( $new_instance, $old_instance ) {
...
-					'tabSize'    => 2,
-				),
-			)
-		);
+		if ( ! isset( $this->number ) ) {
+			return;
+		}
 
+		wp_enqueue_code_editor( array( 'type' => 'text/html' ) );
+		wp_enqueue_script( 'wp-codemirror' );
```
Comment (mattyrob) [inline src/wp-includes/widgets/class-wp-widget-custom-html.php:207]: I think this should be a condition enqueue based on user level setting, I can push a change.

**Review by mattyrob — CHANGES_REQUESTED** [review]

**Comment by xxsimoxx** [comment]
Seems that the assertion failing in `test__register` can be removed and `Tests_Widgets_wpWidgetCustomHtml::test_render_control_template_scripts` can be removed.

File: src/wp-includes/widgets/class-wp-widget-custom-html.php (hunk 1 above)
Comment (KTS915) [reply src/wp-includes/widgets/class-wp-widget-custom-html.php:218]: Go ahead and make that change along with the one below that you've suggested, @mattyrob .

**Comment by mattyrob** [comment]
It looks like some more recent changes reverted this commit:
https://github.com/ClassicPress/ClassicPress/pull/1628/commits/d6d5a97a7f6c9b6973fe7c672448321434397c88

This is resulting in the failing tests, can that change be restored? Or was that an intentional reversion?

**Comment by KTS915** [comment]
That check should certainly be restored. Good catch!

**Review by xxsimoxx — APPROVED** [review]

**Review by mattyrob — APPROVED** [review]

**Comment by mattyrob** [comment]
Looks okay based on code, some testing though shows duplicate content:
Screenshot 2024-11-06 at 16 28 32

**Comment by KTS915** [comment]
Can you try it with PR #1631 activated too? I think that together they resolve each other's duplication issue.

**Pull request merged**